pkg/agent/multicluster/stretched_networkpolicy_controller.go

```diff
@@ -0,0 +1,426 @@
+	containerConfigs := s.interfaceStore.GetContainerInterfacesByPod(classifierFlowUpdate.PodRef.Name, classifierFlowUpdate.PodRef.Namespace)
+	if len(containerConfigs) == 0 {
+		return fmt.Errorf("pod container config not found")
+	}
+	return s.ofClient.UpdatePodClassifierFlowsWithNewLabelID(
+		containerConfigs[0].InterfaceName,
+		classifierFlowUpdate.LabelID,
+		uint32(containerConfigs[0].OFPort),
+		containerConfigs[0].IPs,
+	)
+}
+
+func (s *StretchedNetworkPolicyController) getClassifierFlowUpdate(podRef types.NamespacedName, normalizedLabel string) *classifierFlowInfo {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	return s.getClassifierFlowUpdateLocked(podRef, normalizedLabel)
+}
+
+func (s *StretchedNetworkPolicyController) getClassifierFlowUpdateLocked(podRef types.NamespacedName, normalizedLabel string) *classifierFlowInfo {
+	oldNormalizedLabel, ok := s.podToLabel[podRef]
+	if ok && oldNormalizedLabel != normalizedLabel {
+		s.deleteLabelToPod(oldNormalizedLabel, podRef)
+	}
+	if !ok || oldNormalizedLabel != normalizedLabel {
+		s.addLabelToPod(normalizedLabel, podRef)
+		s.podToLabel[podRef] = normalizedLabel
+	}
+	labelID := openflow.UnknownLabelIdentity
+	if id, ok := s.labelIdentityCache[normalizedLabel]; ok {
+		labelID = id
+	}
+	return &classifierFlowInfo{
+		PodRef:  podRef,
+		LabelID: labelID,
+	}
+}
+
+func (s *StretchedNetworkPolicyController) processPodCNIAddEvent(e interface{}) {
+	podEvent := e.(antreatypes.PodUpdate)
+	if !podEvent.IsAdd {
+		return
+	}
+	podRef := types.NamespacedName{
+		Namespace: podEvent.PodNamespace,
+		Name:      podEvent.PodName,
+	}
+	pod, err := s.podLister.Pods(podRef.Namespace).Get(podRef.Name)
+	if err != nil {
+		klog.ErrorS(err, "Can't get Pod", "name", podRef.Name, "namespace", podRef.Namespace)
+		return
+	}
+	if pod.Spec.HostNetwork {
+		klog.V(2).InfoS("Received a Pod add event but decided to skip",
+			"name", pod.Name, "namespace", pod.Namespace, "node", pod.Spec.NodeName, "hostNetwork", pod.Spec.HostNetwork)
+		return
+	}
+	podNS, err := s.namespaceLister.Get(pod.Namespace)
+	if err != nil {
+		klog.ErrorS(err, "Can't get the Namespace of a Pod", "name", pod.Name, "namespace", pod.Namespace)
+		return
+	}
+	normalizedLabel := multicluster.GetNormalizedLabel(podNS.Labels, pod.Labels, podNS.Name)
+	classifierFlowUpdate := s.getClassifierFlowUpdate(podRef, normalizedLabel)
+	s.queue.Add(classifierFlowUpdate)
+}
+
+// processPodUpdate handles Pod update events. It only enqueues the Pod if the
+// Labels of this Pod has been updated.
+func (s *StretchedNetworkPolicyController) processPodUpdate(old, cur interface{}) {
+	oldPod, oldIsPod := old.(*v1.Pod)
+	curPod, curIsPod := cur.(*v1.Pod)
+	if !oldIsPod || !curIsPod {
+		klog.ErrorS(errors.New("received unexpected object"), "Pod UpdateFunc can't process event", "obj", cur)
+		return
+	}
+	if curPod.Spec.HostNetwork {
+		klog.InfoS("received a Pod update event but decided to skip",
```

Comment:
It should be clear why it decide to skip, and this may not worth default level logging as it's not really helpful to know hostNetwork Pod is skipped.
```
klog.V(5).InfoS("Skipped processing hostNetwork Pod", ...)
```

Reply:
Done.